veerer/linear_family.py

```diff
@@ -213,6 +220,69 @@ def __init__(self, *args, mutable=False, check=True):
         if check:
             self._check(ValueError)
 
+    def __getstate__(self):
+        R = self.base_ring()
+        a = list(self._fp[:])
+        a.extend(self._ep)
+        a.extend(self._colouring)
+        a.append(self._mutable)
+        # NOTE: here we know that all entries have the same parent
+        if R is ZZ:
+            entries = list(map(int, self._subspace.list()))
+        elif R is QQ:
+            entries = []
+            for x in self._subspace.list():
+                entries.append(int(x.numerator()))
+                entries.append(int(x.denominator()))
+        elif _NumberFields is not None and R in _NumberFields:
+            entries = []
+            for x in self._subspace.list():
+                v = x.vector()
+                d = v.vector().denominator()
+                entries.extend(int(x) for x in (v * d))
+                entries.append(int(d))
+        else:
+            entries = self._subspace.list()
+        return a, R, entries
+
+    def __setstate__(self, arg):
+        r"""
+        TESTS::
+
+            sage: from veerer import VeeringTriangulation
+            sage: vt = VeeringTriangulation("(0,1,2)(~0,~1,~2)", "RRB")
+            sage: lf = vt.as_linear_family()
+            sage: lf2 = loads(dumps(lf))  # indirect doctest
+            sage: assert lf == lf2
+            sage: lf2._check()
+        """
+        a, R, raw_entries = arg
+        n = (len(a) - 1) // 3
+        self._n = n
+        self._fp = array('i', a[:n])
+        self._ep = array('i', a[n:2*n])
+        self._colouring = array('i', a[2*n:3*n])
+        self._mutable = a[-1]
+        self._vp = array('i', [-1] * n)
+        for i in range(n):
+            self._vp[self._fp[self._ep[i]]] = i
+
+        if R is ZZ:
+            entries = [ZZ(x) for x in raw_entries]
+        elif R is QQ:
+            entries = []
+            for i in range(0, len(raw_entries), 2):
+                entries.append(QQ((raw_entries[i], raw_entries[i + 1])))
+        elif _NumberFields is not None and R in _NumberFields:
+            entries = []
+            d = R.degree()
+            for i in range(0, len(raw_entries), d + 1):
+                v = tuple(x / raw_entries[i + d] for x in raw_entries[i:i + d])
+                entries.append(R(v))
```

Comment:
Isn't there an `else: entries = raw_entries` missing?

Reply:
Don't think so, there is a `entries = []` before the loop and then it is populated from `raw_entries`.

Reply:
I will at least add a doctest to it!

Reply:
I understood what you meant :-) (but I do not know how to test it though)

Reply:
fixed in 2c3184a. Thanks.